### PR TITLE
GrafanaUI: Add minWidth and maxWidth props to the Box component

### PR DESCRIPTION
**Pull request by konrad147**
**What is this feature?**
This PR adds `minWidth` and `maxWidth` props to the `Box` component

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Layout/Box/Box.tsx [hunk 1]
```diff
@@ -58,6 +59,10 @@ interface BoxProps extends FlexProps, Omit<React.HTMLAttributes<HTMLElement>, 'c
   justifyContent?: ResponsiveProp<JustifyContent>;
   gap?: ResponsiveProp<ThemeSpacingTokens>;
 
+  // Size props
+  minWidth?: ResponsiveProp<Property.MinWidth>;
+  maxWidth?: ResponsiveProp<Property.MaxWidth>;
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/Layout/Box/Box.tsx:64]: Can we also add width and height at the same time?

File: packages/grafana-ui/src/components/Layout/Box/Box.tsx [hunk 2]
```diff
@@ -290,6 +301,12 @@ const getStyles = (
       getResponsiveStyle(theme, gap, (val) => ({
         gap: theme.spacing(val),
       })),
+      getResponsiveStyle(theme, minWidth, (val) => ({
+        minWidth: val,
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/Layout/Box/Box.tsx:305]: Wonder if this should be a value for the `spacing` function instead like it's done in other components 🤔

File: packages/grafana-ui/src/components/Layout/Box/Box.tsx (hunk 1 above)
Comment (Clarity-89) [inline packages/grafana-ui/src/components/Layout/Box/Box.tsx:64]:
```suggestion
  // Size props
  minWidth?: ResponsiveProp<Property.MinWidth<number>>;
  maxWidth?: ResponsiveProp<Property.MaxWidth<number>>;
  width?: ResponsiveProp<Property.Width<number>>;
  height?: ResponsiveProp<Property.Height<number>>;
```

File: packages/grafana-ui/src/components/Layout/Box/Box.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (konrad147) [reply packages/grafana-ui/src/components/Layout/Box/Box.tsx]: It seems we don't need the typecheck here. `theme.spacing` handles strings correctly

Comment (Clarity-89) [reply packages/grafana-ui/src/components/Layout/Box/Box.tsx]: Yeah, the rationale for using it is that if in the future we decide to switch to using sizing tokens, it'll be easier if the props are already aligned with the spacing function.

**Review by Clarity-89 — APPROVED** [review]
Looks good! 🚀  Thanks for taking care of this!

**Pull request merged**